glib/src/subclass/object.rs

```diff
@@ -133,6 +133,52 @@ unsafe extern "C" fn dispose<T: ObjectImpl>(obj: *mut gobject_ffi::GObject) {
     }
 }
 
+// rustdoc-stripper-ignore-next
+/// Trait containing only the property related functions of `ObjectImpl`.
+/// Implemented by the `Props` macro.
+/// When implementing `ObjectImpl` you may want to delegate the function calls to this trait.
+/// Some functions have a different return type from `ObjectImpl` to handle possible errors caused by missing
+/// properties.
+pub trait AutoObjectProps: ObjectSubclass {
+    // rustdoc-stripper-ignore-next
+    /// Properties installed for this type.
+    fn properties() -> &'static [ParamSpec] {
+        &[]
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Similar to [`ObjectImpl`](trait.ObjectImpl.html) but returns a boolean, `true` in the case
+    /// a property was set correctly, `false` if the property wasn't set
+    fn set_property(
+        &self,
+        _obj: &Self::Type,
+        _id: usize,
+        _value: &Value,
+        _pspec: &ParamSpec,
+    ) -> bool {
+        false
+    }
+
+    // rustdoc-stripper-ignore-next
+    /// Similar to [`ObjectImpl`](trait.ObjectImpl.html) but returns an `Option<Value>`,
+    /// `None` in the case the property wasn't found
+    fn property(&self, _obj: &Self::Type, _id: usize, _pspec: &ParamSpec) -> Option<Value> {
```

Comment:
It should just panic if the property was not found or not? It would've been an error to call it in that case

Reply:
I'd prefer the panic to happen in `ObjectImpl::property`, not here. It's more explicit and more flexible.